lib/cc/analyzer/engine_output_filter.rb

```diff
@@ -0,0 +1,40 @@
+module CC
+  module Analyzer
+    class EngineOutputFilter
+      ISSUE_TYPE = "issue".freeze
+
+      def initialize(config = {})
+        @config = config
+      end
+
+      def filter?(output)
+        if (json = parse_as_json(output))
+          issue?(json) && ignore_issue?(json)
+        else
+          false
+        end
+      end
+
+      private
+
+      def parse_as_json(output)
+        JSON.parse(output)
+      rescue JSON::ParserError
+        nil
+      end
+
+      def issue?(json)
+        json.is_a?(Hash) && json["type"] == ISSUE_TYPE
```

Comment:
That's valid for JSON, but not our spec, right? Is there a situation where an engine will output an array?